gm4_tnt_landmines/data/tnt_landmines/functions/main.mcfunction

```diff
@@ -0,0 +1,19 @@
+#creation
+execute as @e[type=item,nbt={OnGround:1b,Item:{id:"minecraft:player_head",Count:1b,tag:{gm4_tnt_landmines:{item:1}}}},tag=!gm4_mine_safe] at @s unless entity @e[type=armor_stand,tag=gm4_tnt_mine,distance=..1.5] unless entity @e[type=item,nbt={OnGround:1b,Item:{id:"minecraft:player_head",Count:1b,tag:{gm4_tnt_landmines:{item:1}}}},distance=0.1..1.5] run function tnt_landmines:create
+execute as @e[type=armor_stand,tag=gm4_tnt_mine,tag=!gm4_armed] at @s unless entity @e[type=area_effect_cloud,tag=gm4_mine_disarmed,distance=..0.2] run tag @s add gm4_armed
+
+#upgrading
+execute as @e[type=item,nbt={OnGround:1b,Item:{id:"minecraft:gunpowder",Count:1b,tag:{gm4_tnt_landmines:{item:2}}}}] at @s unless entity @e[type=item,nbt={OnGround:1b,Item:{id:"minecraft:gunpowder",Count:1b,tag:{gm4_tnt_landmines:{item:2}}}},distance=0.1..1.1] at @e[type=armor_stand,tag=gm4_tnt_mine,distance=..1.2,limit=1,tag=!gm4_no_damage] unless entity @e[type=area_effect_cloud,tag=gm4_mine_disarmed,distance=..0.2] run function tnt_landmines:upgrade
```

Comment:
Both for upgrading and disarming you should check this in the function mentioned above. This way you can use `distance` to prevent the nbt of all items being checked.